server/src/internalClusterTest/java/org/opensearch/action/admin/cluster/node/tasks/ConcurrentSearchTasksIT.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ *
+ * The OpenSearch Contributors require contributions made to
+ * this file be licensed under the Apache-2.0 license or a
+ * compatible open source license.
+ */
+
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+/*
+ * Modifications Copyright OpenSearch Contributors. See
+ * GitHub history for details.
+ */
+
+package org.opensearch.action.admin.cluster.node.tasks;
+
+import org.opensearch.action.admin.indices.segments.IndicesSegmentsRequest;
+import org.opensearch.common.settings.FeatureFlagSettings;
+import org.opensearch.common.settings.Setting;
+import org.opensearch.common.settings.Settings;
+import org.opensearch.common.util.FeatureFlags;
+
+/**
+ * Integration tests for task management API with Concurrent Segment Search
+ *
+ * The way the test framework bootstraps the test cluster makes it difficult to parameterize the feature flag.
+ * Once concurrent search is moved behind a cluster setting we can parameterize these tests behind the setting.
+ */
+public class ConcurrentSearchTasksIT extends TasksIT {
```

Comment:
Please use `OpenSearchIntegTestCase` as the base class, there is no need to rerun all `TasksIT` again since there is only a single test case you need to run: testConcurrentSearchTaskTracking

Reply:
You could create parametrized ConcurrentSearchTasksIT to test different scenarios, I don't see how the existing tests in the TaskIT are affected by the changes related to concurrent search.

Reply:
Fair enough, I think the other tests in `TaskIT` also don't index enough docs or create enough segments to go down the concurrent search path regardless. Will make the change.

Reply:
@reta I pulled out a lot of the helpers into a new `AbstractTasksIT` to avoid duplicating the TaskManagerListener logic and so that it's easy to add more `ConcurrentSearchTasksIT` tests as a part of the API response changes if needed. Let me know what you think!